[CONTRIB] Create custom expectation for Chi Square Test

HaebichanGX opened the pull request:
Create custom expectation for calculating and evaluating p-value for a chi square test based on two columns in a dataset




- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

HaebichanGX committed:
Create custom expectation for Chi Square

Create custom expectation for calculating and evaluating p-value for a chi square test based on two columns in a dataset

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8314&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/8314/526f7193/ff5fd15555046fd65378c3f6dd689894f0a7a4ef.svg)](https://app.codesee.io/r/reviews?pr=8314&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

HaebichanGX committed:
Make lint changes

Make lint changes as PR request

HaebichanGX committed:
Fix pylint import sorting error

Trying to fix this error in lint: "contrib/experimental/great_expectations_experimental/expectations/expect_column_chisquare_test_pvalue_to_be_greater_than.py:7:1: I001 [*] Import block is un-sorted or un-formatted"

TrangPham commented on contrib/experimental/great_expectations_experimental/expectations/expect_column_chisquare_test_pvalue_to_be_greater_than.py:
```diff
@@ -0,0 +1,159 @@
+"""
+This is a template for creating custom BatchExpectations.
```
Could you update or remove this comment block?

HaebichanGX replied:
sounds good

TrangPham commented on contrib/experimental/great_expectations_experimental/expectations/expect_column_chisquare_test_pvalue_to_be_greater_than.py:
```diff
@@ -0,0 +1,159 @@
+
+class ColumnChisquareTestPValueGreaterThan(TableMetricProvider):
+    # This is the id string that will be used to reference your Metric.
+    metric_name = "column.p_value_meets_threshold"
```
Nit: class name uses "greater than" while metric name uses "meets threshold". Could the metric name be updated to use "greater than" as well?

HaebichanGX replied:
sounds good, done.

HaebichanGX committed:
Make changes per Thu's recommendation

Make naming improvements, remove doc strings that don't need to be there.

HaebichanGX committed:
Merge branch 'develop' into add_chi_squared_custom_expectation

github-actions[bot] committed:
Merge develop into add_chi_squared_custom_expectation

HaebichanGX committed:
Merge branch 'develop' into add_chi_squared_custom_expectation

github-actions[bot] committed:
Merge develop into add_chi_squared_custom_expectation
